src/openforms/submissions/models.py

```diff
@@ -236,6 +236,10 @@ class Submission(models.Model):
         ),
     )
 
+    previous_submission = models.ForeignKey(
+        "submissions.Submission", on_delete=models.DO_NOTHING, null=True, blank=True
```

Comment:
you really should never use `models.DO_NOTHING` because then your database will start tripping up. Since the field is nullable, `models.SET_NULL` is more appropriate.

Effectively this means that when a submission is deleted (which happens after x number of days!), any change-appointment submissions now have lost their reference. I think this is acceptable.